packages/2d/src/components/SVG.ts

```diff
@@ -0,0 +1,387 @@
+    return color;
+  }
+
+  private getElementStyle(
+    element: SVGGraphicsElement,
+    inheritedStyle: ShapeProps,
+  ): ShapeProps {
+    return {
+      fill:
+        this.parseColor(element.getAttribute('fill')) ?? inheritedStyle.fill,
+      stroke:
+        this.parseColor(element.getAttribute('stroke')) ??
+        inheritedStyle.stroke,
+      lineWidth: element.hasAttribute('stroke-width')
+        ? parseFloat(element.getAttribute('stroke-width')!)
+        : inheritedStyle.lineWidth,
+      layout: false,
+    };
+  }
+
+  private applyTransformToShape(shape: Shape, transform: DOMMatrix) {
+    const position = {
+      x: transform.m41,
+      y: transform.m42,
+    };
+    const rotation = (Math.atan2(transform.m12, transform.m11) * 180) / Math.PI;
+    const scale = {
+      x: Vector2.magnitude(transform.m11, transform.m12),
+      y: Vector2.magnitude(transform.m21, transform.m22),
+    };
+    const determinant =
+      transform.m11 * transform.m22 - transform.m12 * transform.m21;
+    if (determinant < 0) {
+      if (transform.m11 < transform.m22) scale.x = -scale.x;
+      else scale.y = -scale.y;
+    }
+    shape.position(position);
+    shape.rotation(rotation);
+    shape.scale(scale);
+  }
+
+  private *extractGroupShape(
+    element: SVGElement,
+    svgRoot: Element,
+    parentTransform: DOMMatrix,
+    inheritedStyle: ShapeProps,
+  ): Generator<Node> {
+    for (const child of element.children) {
+      if (!(child instanceof SVGGraphicsElement)) continue;
+
+      yield* this.extractElementShape(
+        child,
+        svgRoot,
+        parentTransform,
+        inheritedStyle,
+      );
+    }
+  }
+
+  private *extractElementShape(
+    child: SVGGraphicsElement,
+    svgRoot: Element,
+    parentTransform: DOMMatrix,
+    inheritedStyle: ShapeProps,
+  ): Generator<Node> {
+    const transformMatrix = this.getElementTransformation(
+      child,
+      parentTransform,
+    );
+    const style = this.getElementStyle(child, inheritedStyle);
+    if (child.tagName == 'g')
+      yield* this.extractGroupShape(child, svgRoot, transformMatrix, style);
+    else if (child.tagName == 'use') {
+      const hrefElement = svgRoot.querySelector(
+        (child as SVGUseElement).href.baseVal,
+      )!;
+      if (!(hrefElement instanceof SVGGraphicsElement)) return;
+
+      yield* this.extractElementShape(
+        hrefElement,
+        svgRoot,
+        transformMatrix,
+        inheritedStyle,
+      );
+    } else if (child.tagName == 'path') {
+      const data = child.getAttribute('d');
+      if (!data) {
+        useLogger().warn('blank path data at ' + child.id);
+        return;
+      }
+      const path = new Path({
+        data,
+        ...style,
+      });
+      const center = path.getPathBBox().center;
+      this.applyTransformToShape(
+        path,
+        transformMatrix.translate(center.x, center.y),
+      );
+      yield path;
+    } else if (child.tagName == 'rect') {
```

Comment:
What about `circle`, `ellipse` and `image`?

Reply:
This PR only focus on Latex tweening. Latex SVG does not contain `cirlce`, `ellipse` or `image`. In my opinion, implementing shape other than used in Latex must be in another PR.